server/SessionService.java

```diff
@@ -53,7 +53,11 @@ public SessionService(TypeDBService typeDBSvc, TypeDB.Session session, Options.S
         this.accessLock = new StampedLock().asReadWriteLock();
         this.isOpen = new AtomicBoolean(true);
         this.transactionServices = new ConcurrentSet<>();
-        mayStartIdleTimeout();
+        startIdleTimeout();
+    }
+
+    private synchronized void startIdleTimeout() {
+        idleTimeoutTask = scheduled().schedule(this::idleTimeout, options.sessionIdleTimeoutMillis(), MILLISECONDS);
```

Comment:
would expect to find this function near the other methods to do with starting/stopping idle timeouts

Reply:
I definitely want this method to be right after the constructor to make it crystal clear that `idleTimeoutTask` is initialised at construction and therefore never null. I considered pulling the rest of the related functions up to group with this, but that would push the accessors further down than I'd like.

Reply:
hmm I think it's clear enough in the constructor and we can put `startIdleTimeout` near the others, that is the stronger grouping